### ISSUE TITLE
[Feature Request] Add custom metadata to a collection 

### ISSUE THREAD

**Issue by oankarberg** [typesense/typesense#1246]
## Description
Just a nice to have feature. 
Have the ability to add custom `metadata` to typesense collections. 

It could be some of but not limited to:  
* Which batch job created this collection initially.
* How far back does the collection contain indexed data. 
* Other identifiers that can describe the indexed data / purpose of the collection. 

## Expected Behavior
Introduce some field, for example `metadata`, with the type `object` on a `collection` that can contain `key: value`  pairs. 

The metadata would show up in the collection settings from `/api/collections` endpoint 

```js
{
  "created_at": 1689756292,
  "default_sorting_field": "",
  "enable_nested_fields": true,
  "fields": [],
  "name": "collection_name",
  "num_documents": 123456,
  "symbols_to_index": [],
  "token_separators": [],
  // new field
  "metadata": {
    "batch_job":"",
    "indexed_from":"2023-04-20T00:00:00.000Z"
  }
}
```

**Comment by clarson99** [comment]
I agree, it would be very useful to have metadata on both collections and fields.  

Having metadata on the fields would be very convenient for cases where the UI is being generated dynamically from the schema. For instance, some facets are more important than others and should always be shown or sorted to the top.  Another use case could be using metadata to determine if a field should be shown in the list of initial search results, or if that field should be shown on the "more details" page/panel/modal.

**Comment by kishorenc** [comment]
Feature available in `0.26.0.rc48`.

You can use the `metadata` key to add additional information to collection schema:

```
 curl -k "http://localhost:8108/collections" -X POST -H "Content-Type: application/json"  -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" -d '{
            "name": "hnstories",
            "enable_nested_fields": true,
            "fields": [
              {"name": "title", "type": "string", "facet":true },
              {"name": "points", "type": "int32" }
            ],
            "metadata":{
             "batch_job":325,
             "indexed_from":"2023-04-20T00:00:00.000Z"
          }'
```

**Comment by jasonbosco** [comment]
Released in v26
